tools/perf/ucc_pt_coll_reduce_scatter.cc

```diff
@@ -0,0 +1,92 @@
+#include "ucc_pt_coll.h"
+#include "ucc_perftest.h"
+#include <ucc/api/ucc.h>
+#include <utils/ucc_math.h>
+#include <utils/ucc_coll_utils.h>
+
+ucc_pt_coll_reduce_scatter::ucc_pt_coll_reduce_scatter(ucc_datatype_t dt,
+                        ucc_memory_type mt, ucc_reduction_op_t op,
+                        bool is_inplace,
+                        ucc_pt_comm *communicator) : ucc_pt_coll(communicator)
+{
+    has_inplace_   = true;
+    has_reduction_ = true;
+    has_range_     = true;
+    has_bw_        = true;
+
+    coll_args.coll_type = UCC_COLL_TYPE_REDUCE_SCATTER;
+    coll_args.mask = 0;
+    if (is_inplace) {
+        coll_args.mask = UCC_COLL_ARGS_FIELD_FLAGS;
+        coll_args.flags = UCC_COLL_ARGS_FLAG_IN_PLACE;
+    }
+
+    coll_args.mask |= UCC_COLL_ARGS_FIELD_PREDEFINED_REDUCTIONS;
+    coll_args.reduce.predefined_op = op;
+    coll_args.src.info.datatype = dt;
+    coll_args.src.info.mem_type = mt;
+    coll_args.dst.info.datatype = dt;
+    coll_args.dst.info.mem_type = mt;
+}
+
+ucc_status_t ucc_pt_coll_reduce_scatter::init_coll_args(size_t count,
+                                                        ucc_coll_args_t &args)
+{
+    size_t size;
+    ucc_status_t st;
+
+    if (count % comm->get_size() != 0) {
```

Comment:
@Sergei-Lebedev i was running this PR with tl/ucp reduce-scatter and faced a problem that the test is effectively disabled for many cases. E.g., running on 1 jazz node with ppn=28. It would be skipped due to this condition.

I suggest we change it this way:
if (count < comm->get_size()) {
    skip;
}
count = count - (count % comm->get_size());

Reply:
@vspetrov this expected, size should be multiple of number of ranks. Maybe for reduce_scatter and scatter make count to be number of elements to receive?

Reply:
sure, that is essentially the same. i'm just saying lets not disable test for odd rank counts. we can make count = count * comm->get_size()

Reply:
fixed